test/functional/feature_init.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Stress tests related to node initialization."""
+import random
+import time
+import os
+from pathlib import Path
+
+from test_framework.test_framework import BitcoinTestFramework, SkipTest
+from test_framework.test_node import ErrorMatch
+from test_framework.util import assert_equal
+
+
+class InitStressTest(BitcoinTestFramework):
+    """
+    Ensure that initialization can be interrupted at a number of points and not impair
+    subsequent starts.
+    """
+
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+
+    def run_test(self):
+        """
+        - test bailing initialization after seeing a certain log line.
+        - test bailing init after seeing a random number of log lines.
+        - test removing certain essential files to test startup error paths.
+        """
+        # TODO: skip Windows for now since it isn't clear how to SIGTERM.
+        #
+        # Windows doesn't support `process.terminate()`.
+        # and other approaches (like below) don't work:
+        #
+        #   os.kill(node.process.pid, signal.CTRL_C_EVENT)
+        if os.name == 'nt':
+            raise SkipTest("can't SIGTERM on Windows")
+
+        self.stop_node(0)
+        node = self.nodes[0]
+
+        def sigterm_node():
+            node.process.terminate()
+            node.process.wait()
+            node.debug_log_path.unlink()
```

Comment:
I think this was not addressed?

Reply:
This suggestion is at odds with the nature of the test itself, since the logfile contents determine when to sigterm and so have to be cleared each iteration. 

If someone would like to improve this test by consuming the node's stdout instead of the logfile, that's one possible improvement suitable for a follow-up.

Reply:
Seems odd to delete a file to allow reading it. Why can´t this use the same approach like assert_debug_log, that is remember the last read position?